CPP_STYLE_GUIDE.md

```diff
@@ -0,0 +1,120 @@
+# C++ style guide
+
+Unfortunately, the C++ linter (based on
+[Google’s `cpplint`](https://github.com/google/styleguide)), which can be run
+explicitly via `make cpplint`, does not currently catch a lot of rules that are
+specific to the Node.js C++ code base. This document explains the most common of
+these rules:
+
+## Left-leaning (C++ style) asterisks for pointer declarations
+
+`char* buffer;` instead of `char *buffer;`
+
+## 2 spaces of indentation for blocks or bodies of conditionals
+
+```c++
+if (foo)
+  bar();
+```
+
+or
+
+```c++
+if (foo) {
+  bar();
+  baz();
+}
+```
+
+(Braces are optional if the statement body only has one line.)
+
+## 4 spaces of indentation for statement continuations
+
+```c++
+VeryLongTypeName very_long_result = SomeValueWithAVeryLongName +
+    SomeOtherValueWithAVeryLongName;
```

Comment:
the linter doesn't currently catch whether the operation is at the end of the line or the beginning of the next line. couple examples:
```cpp
// ternary
int r = true ?
    1 : 0;
int r = true
    ? 1 : 0;

// conditional
if (foo &&
    bar) { }
if (foo
    && bar) { }
```
specify this?

@addaleax   /cc @bnoordhuis

Reply:
The `&&` should go on the same line as the expression preceding it.

Ternary operator: we don't have a hard rule, I think, but IMO if it doesn't fit on a single line, you should be using an `if` statement anyway.

Reply:
re: placement of `&&`. linter doesn't check that. should be possible to add to the linter? if so then no reason to put it here.